### DISCUSSION TITLE
Import old transactions. Best flow how to to it?

### DISCUSSION THREAD

**Discussion by thehijacker** [firefly-iii/firefly-iii#9355] [Firefly III]
I have started using Firefly in november 2023. Set the current state of the accounts and started from there. My current state is up to date and all is fine. Now I have found out I can export from my online bank all records going back to 2015. So my idea is to import them so I have history.

For my wife I done the same since she started using Firefly few months ago. Exported transactions for few years and imported them via Firefly imported. All worked fine but this was fresh new account.

How to handle my situation? Importing past transaction and still keeping current state untouched.

Thanks.

**Comment by JC5** [comment]
The best way to do this is to import them month by month, going backwards in time. 

What you will run into potentially is that your opening balance will be wrong, so you have to correct it every time, to see if the import succeeded. Other than that, it should work.

**Reply by thehijacker** [reply]
And done. I have imported 4696 additional transactions. Used CLI method as the web one gave me timeout so I restored database and used CLI method. When done my actual state was of course wrong. So I edited opening balance for the account in a way that my current account state is correct.

Took a while to import, since rules were applied to most of them. Should number of transactions impact the GUI speed? I have a feeling it is a bit slower now.

I do have lots of new transactions now that have a **(no name)** under target account. Seems to be for the imported transaction that I do not set target account from rules. I need to create the new rules and apply them. Tried it on once and it took a long long time to apply this for over 900 transactions.

What can I do to speed it up? Put MySQL database on SSD disk? Add more CPU or RAM to the Firefly container in Proxmox?

**Reply by JC5** [reply]
> What can I do to speed it up? Put MySQL database on SSD disk? Add more CPU or RAM to the Firefly container in Proxmox?

Set all rules to "stop processing". If they are executed, other rules in the same group are skipped. If you order your rules smartly, you should see some speed improvements.
